pkg/services/ssosettings/validation/oauth_validators.go

```diff
@@ -17,6 +17,20 @@ func AllowAssignGrafanaAdminValidator(info *social.OAuthInfo, requester identity
 	return nil
 }
 
+func OrgMappingValidator(info *social.OAuthInfo, requester identity.Requester) error {
+	if len(info.OrgMapping) > 0 && !requester.GetIsGrafanaAdmin() {
```

Comment:
If `orgMapping` is configured for a provider then only Grafana Admins will be able to edit the configuration. What I think we can do instead is to check whether non Grafana admins are modifying the existing `orgMapping`. The SSO validation function contains the `oldSettings` parameter so we can check if `oldSettings.orgMapping` is the same with `currentSettings.orgMapping`.

Reply:
Done